SNTP client library over sockets.

rveerama1 opened the pull request:
Migrated net_app based SNTP client library to socket based library and deleted net_app based library.

rveerama1 committed:
net: samples: Add socket based SNTP client application

SNTP client sample requests time sync on IPv4 and IPv6 addresses
using socket based SNTP client library.

Signed-off-by: Ravi kumar Veeramally <ravikumar.veeramally@linux.intel.com>

tbursztyka commented on include/net/sntp_sock.h:
```diff
@@ -0,0 +1,91 @@
+	sntp_resp_cb_t      cb;
+
+	/** The user data of SNTP response callback */
+	void                *user_data;
```
No need of that. If the one wants some data, up to the struct sntp_ctx to be part of a private struct, and CONTAINER_OF() on the ctx would do the trick.

tbursztyka commented on include/net/sntp_sock.h:
```diff
@@ -0,0 +1,91 @@
+int sntp_request(struct sntp_ctx *ctx,
+		 u32_t timeout,
+		 sntp_resp_cb_t callback,
+		 void *user_data);
```
So you could remove that parameter

tbursztyka commented on include/net/sntp_sock.h:
```diff
@@ -0,0 +1,91 @@
+ *
+ * @param ctx Address of sntp context.
+ * @param timeout Timeout of sending sntp request (in milliseconds).
+ * @param callback Callback function of sntp response.
```
Detail that it must be valid and not NULL

tbursztyka commented on subsys/net/lib/sntp_sock/sntp_sock.c:
```diff
@@ -0,0 +1,232 @@
+	}
+
+	status = parse_response((u8_t *)&buf, sizeof(buf),
+				sntp->expected_orig_ts, &tmp);
```
I don't think you need tmp at all, just use epoch_time.

If status is not 0, user should not care about epoch_time validity.

tbursztyka commented on subsys/net/lib/sntp_sock/sntp_sock.c:
```diff
@@ -0,0 +1,232 @@
+		return -EINVAL;
+	}
+
+	ctx->cb = callback;
```
NULL cb should not be allowed, or then user will never get the result of the request.

tbursztyka commented on subsys/net/lib/sntp_sock/sntp_sock.c:
```diff
@@ -0,0 +1,232 @@
+}
+
+static s32_t parse_response(u8_t *data, u16_t len, u32_t orig_ts,
+				u64_t *epoch_time)
```
parameter alignement, it should be below last '(' + 1 space.

And return int, not s32_t

nashif reviewed (dismissed):
Not fond of this approach of moving to sockets. We should find a way to still use the same headers and same structure and still provide a deprecation path. Look at 

https://github.com/nashif/zephyr/commit/50bfcb8d5e4924a03dd504fa8773fad6cfdd6dfb for CoAP and dealing with Kconfigs.

For the headers, I would keep the sntp.h and add a check in the header for the right implementation to pick instead of introducing a parallel universe. We have done this in the past, and it failed miserably.

tbursztyka commented on include/net/sntp_sock.h:
```diff
@@ -0,0 +1,91 @@
+	void                *user_data;
+
+	/** Is this context setup or not */
+	bool                is_init;
```
I don't think you need that, if it's not initialized fd will be invalid and sock operation should fail.

tbursztyka commented on subsys/net/lib/sntp_sock/sntp_sock.c:
```diff
@@ -0,0 +1,232 @@
+		return -EINVAL;
+	}
+
+	if (epoch_time) {
```
no need to test this, it must be valid. Or then there is not much point of parsing the response I guess.

rveerama1 commented:
@tbursztyka addressed your comments, please take a look.

pfalcon reviewed (dismissed):
Well, I'm trying to be easy with sockets conversion, good that people actually do that.

But some questions has to be asked: why to we have that callback here? The processing is absolutely synchronous, poll() is used only for timeout (someone should submit a feature request to implement timeouts on socket level). So, we can as well just return result from sntp_request().

Note that the response may be obvious here, but it *should be given explicitly* - what are the requirements for such a socket conversion: is it API compatibility with the previous implementation, or being done in a native socket way.

And if the requirement was to preserve API compatibility (how would anyone know about that, where's design ticket?), then it's already broken by following @tbursztyka's suggestion of removing user_data. (So, if there was such a requirement, then @rveerama1, you should have argued your point, not just follow the suggestion blindly).

Ping @mike-scott is you have comments here.

tbursztyka commented:
I don't think API compatibility was a requirement at all, it's just superfluous to store some "user data" when user can do that already as I suggested.

pfalcon commented:
> I don't think API compatibility was a requirement at all, it's just superfluous to store some "user data" when user can do that already as I suggested.

I think the same. I told previously, that I consider socket conversion as a good opportunity not just to use more native API for socket-based designs (which is the baseline IMHO), but also to improve the functionality where possible (doesn't apply here).

But again, it's hard to read into other peoples' minds.

tbursztyka commented:
@pfalcon hum I read this like there should not be 2 sntp header, the struct names should be same but rest can be all redefined. Both API enclosed in #ifdef #else #endif and that's it.

rveerama1 commented:
> Both API enclosed in #ifdef #else #endif and that's it.

You will get complains from doxygen about duplicate declarations. Because this #ifdef doesn't work there. I tried. Only trick is to exclude file completely to avoid document duplication errors from sanity.

tbursztyka commented:
@rveerama1 what about ignoring old API's documentation? with /** @cond ignore */ ... old api ... /** @endcond */

rveerama1 commented:
> @rveerama1 what about ignoring old API's documentation? with /** @cond ignore _/ ... old api ... /_* @endcond */

aah you mean something like this.
/** @cond DO_NOT_INCLUDE_WITH_DOXYGEN */
/** @endcond (DO_NOT_INCLUDE_WITH_DOXYGEN) */

tbursztyka commented:
@rveerama1 something like that yes

rveerama1 commented:
> Note that the response may be obvious here, but it _should be given explicitly_ - what are the requirements for such a socket conversion: is it API compatibility with the previous implementation, or being done in a native socket way.

I was trying to migrate existing net_app based stuff to socket based, not really re-writing to pure socket based implementation. I really don't have idea of usecases of these protocols over socket based. So I going to stop this attempt. Hence closing this PR.

pfalcon commented:
@rveerama1 : Why not discuss and go forward with the best (or at least better) solution instead?

pfalcon commented:
@rveerama1: Ok, let me pose another question: would you be ok with me taking your patchset and move it forward? Per Zephyr's change rules, I likely will need to fix up your commits. Authorship would remain yours. I'll add myself to Signed-off-by: line. Thanks.

pfalcon commented:
To parallelize this a bit, ping to @nashif re: this comment: https://github.com/zephyrproject-rtos/zephyr/pull/11141#issuecomment-436578044

So, do you think that SNTP API after conversion to socket implementation should stay the same, or it's ok to change it?

nashif commented:
> To parallelize this a bit, ping to @nashif re: this comment: [#11141 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/11141#issuecomment-436578044)
> 
> So, do you think that SNTP API after conversion to socket implementation should stay the same, or it's ok to change it?

you completely mis-understood my comment. I am not talking about the API level. The way it is being added here is forcing us to create a completely parallel universe with the socket implementation with everything being named XYZ_SOCK and duplicating the same set of Kconfigs.

We should instead try to use the same names and driver the deprecation in a different way that avoids duplicating with new naming scheme, removing old implementation, being stuck with the new name forever.

pfalcon commented:
> you completely mis-understood my comment. I am not talking about the API level. 

Maybe. But you misunderstood me to, because I *do* ask whether we should/can break API compatibility here or not.

Back to your point,

> We should instead try to use the same names and driver the deprecation in a different way that avoids duplicating with new naming scheme, removing old implementation, being stuck with the new name forever.

`-` I have to admit that I still don't understand it well. I will need to dig into it to do that. If we should keep old Kconfig option names, then we also should keep the API intact, and then my question is very relevant.

Anyway, my current question is *only* about breaking API or not. For the rest, I'll do my homework first before asking questions.

dbkinder commented on subsys/net/lib/sntp/Kconfig:
```diff
@@ -5,7 +5,7 @@
 
 menuconfig SNTP
-	bool "SNTP Support"
+	bool "SNTP Support using net_app[Deprecated use SNTP_SOCK]"
```
```suggestion
	bool "SNTP Support using net_app (Deprecated, use SNTP_SOCK instead)"
```

dbkinder commented:
We **can** run the doxygen API doc build with certain ``#define`` assumptions (these are configured in the zephyr.doxyfile.in using the ``PREDEFINED`` parameter), and we have the ``EXCLUDE`` configuration parameter that tells doxygen to ignore specific files (or folders), as was done in this PR.  

There is still though the problem if we want to document variations of APIs that would be selected at compile time: as with the compiler seeing multiple definitions of the same name and complaining, the doc build does as well.

dbkinder reviewed (changes requested):
One wording suggestion

rveerama1 commented:
> @rveerama1: Ok, let me pose another question: would you be ok with me taking your patchset and move it forward? Per Zephyr's change rules, I likely will need to fix up your commits. Authorship would remain yours. I'll add myself to Signed-off-by: line. Thanks.

Ok, fine.

pfalcon commented:
Sorry, this got backlogged with other things, but definitely in the queue for 1.14.

@nashif 

>>     Anyway, my current question is only about breaking API or not. For the rest, I'll do my homework first before asking questions.

> no, you should not break the API, we need to keep the old API and deprecate them, if we are to break the API, then we could just yank them out and start with the new API as the main API and avoid having 2 sets of APIs in parallel.

So, my view of this is that conversion to BSD Sockets indeed opens up a possibility to completely redesign older implementations and improve the API. One good example would be HTTP lib. BUT: The whole point is that there should be proper design phase, with requirements collection, RFC, etc.

We definitely can't convert all existing protocol libs at once (nor we should) like that. And if there're low-hanging fruit for conversion, e.g. simple protocols like SNTP, we can go to convert them before the redesign phase. But then we indeed need to spend due diligence on preserving the existing API instead of breaking, as long as it makes sense. I'd like to emphasize that especially if I get involved, because I can't criticize other folks for reshaking logging and shell, and do just the same without any reflection at all.

So, given that no redesign phase was done for this PR, my preference would be to preserve the exact existing API. This includes reverting back to having `void *user_data` param for sntp_request(). Sorry @tbursztyka .

tbursztyka commented:
@pfalcon ok, fine by me.

mschwan-phytec commented:
@pfalcon Hi! Is there any update on this pull request?

The sample application is still missing a README.md containing instructions on how to build and run it and what output to expect. If needed, I could provide some assistance on that.

pfalcon commented:
@mschwan-phytec: Unfortunately, no. For me, the pending matter to figure out here is @rveerama1's vs @nashif's way of handling legacy net_app SNTP implementation, all that stuff is indeed a chore. There may be a novel solution to it cooking up - just remove legacy stuff. Once that is figured out, finishing this may be a hop away.

nashif commented:
@pfalcon 

> @rveerama1: Ok, let me pose another question: would you be ok with me taking your patchset and move it forward? Per Zephyr's change rules, I likely will need to fix up your commits. Authorship would remain yours. I'll add myself to Signed-off-by: line. Thanks.

what is the status of that? IMO we should drop the old implementation and just have the socket based one.

mschwan-phytec commented:
> @pfalcon
> 
> > @rveerama1: Ok, let me pose another question: would you be ok with me taking your patchset and move it forward? Per Zephyr's change rules, I likely will need to fix up your commits. Authorship would remain yours. I'll add myself to Signed-off-by: line. Thanks.
> 
> what is the status of that? IMO we should drop the old implementation and just have the socket based one.

From an application developer's point of view I'd prefer it this way. Having to differentiate between two different implementations of the same functionality is a bit cumbersome.

pfalcon commented:
@nashif:

> IMO we should drop the old implementation and just have the socket based one.

Well, I did my bit of advocacy for preserving older net_app implementations for 1.14. I actually started to write a long ranty reply to one of your comments, but then thought: "What? I'm arguing for keeping net_app stuff??" Well, I was arguing for doing sustainable maintenance. If we want to make an *exception* and tear away net_app stuff sooner rather than later, well, it'll make everyone's life easier. So, I'm just waiting for actions to happen ;-).

nashif commented:
> I'm just waiting for actions to happen

what actions are you waiting for?

pfalcon commented:
> what actions are you waiting for?

Apparently, removal of old net_app-based implementations of SNTP, HTTP, etc. (At least, some people seem to want to do that, and I didn't seem to be able to dissuade them form that.)

nashif commented:
> Apparently, removal of old net_app-based implementations of SNTP, HTTP, etc. (At least, some people seem to want to do that, and I didn't seem to be able to dissuade them form that.)

but socket based SNTP can be done (this PR) in parallel, no need to wait. @rveerama1 can change this PR and remove the net-app based SNTP and replace it with the socket based.

nashif commented:
so in other words, need to know who is going to follow through on this PR and own it, do not want to have folks wait on eachother

rveerama1 committed:
net: sntp: Rework SNTP client library to use sockets

As networking libraries and protocols are moving to socket
based implementation, reworked SNTP client library to use sockets.

Signed-off-by: Ravi kumar Veeramally <ravikumar.veeramally@linux.intel.com>

pfalcon commented:
> so in other words, need to know who is going to follow through on this PR and own it, do not want to have folks wait on eachother

@rveerama1 abandoned this PR, and I picked up. But I didn't have a chance to follow up with it yet - have my own stuff to work on yet. For me, the "blocking" point is to figure out how you wanted to support legacy net_app stuff (vs how @rveerama1 did it). If we can just write that off by completely removing the old stuff, it'll make everyone's life easier (except for perhaps users'). So, hacking on my stuff, waiting for resolution there.

nashif commented:
> So, hacking on my stuff, waiting for resolution there.

still not clear. Are you going to to take this PR and complete it or not? I do not remember @rveerama1 abandoning this, he basically handed it over to you on your request.

rveerama1 commented:
I proposed some changes in this PR.
Removes net_app based SNTP.
Introduces socket based SNTP.
Sample.
No confusion with SNTP_SOCK and SNTP_NET_PKT like that.

nashif commented:
ok, nvm, I see @rveerama1 submitted an update.

rveerama1 commented:
> The sample application is still missing a README.md containing instructions on how to build and run it and what output to expect. If needed, I could provide some assistance on that.

@mschwan-phytec I would be happy to add if you could provide that.

nashif commented:
> @mschwan-phytec I would be happy to add if you could provide that.

@rveerama1 This should actually be part of this PR, a sample needs to have documentation. So please come up with something :-)

nashif commented:
```
subsys/net/lib/Kconfig:23: 'subsys/net/lib/sntp_sock/Kconfig' does not exist.
```

rveerama1 commented:
> @rveerama1 This should actually be part of this PR, a sample needs to have documentation. So please come up with something :-)

True :). If someone has already something in mind. Why not? ;)

mschwan-phytec commented:
> > @rveerama1 This should actually be part of this PR, a sample needs to have documentation. So please come up with something :-)
> 
> True :). If someone has already something in mind. Why not? ;)

Nothing fancy, but here we go:
```rst
.. _sntp-client-sample:

SNTP client sample
##################

Overview
********

This sample application starts an SNTP client and retrieves the current time in
seconds since 1st January 1900.

Requirements
************

- :ref:`networking_with_qemu`
- SNTP or NTP server running on the host

Building and Running
********************

Build the application like this:

.. zephyr-app-commands::
   :zephyr-app: samples/net/sockets/sntp_client
   :board: qemu_x86
   :goals: build
   :compact: run

After the application was run it issues an SNTP request to the host and waits
for a response. As soon as the response gets received the current epoch time in
seconds as well as the status code of the response (0 on success) will be
printed.
```

rveerama1 commented:
@mschwan-phytec Slightly modified Readme text and added LOG_DBG in sample.

mschwan-phytec commented on samples/net/sockets/sntp_client/README.rst:
```diff
@@ -0,0 +1,36 @@
+********
+
+This sample is a simple SNTP client showing how to retrieve the current
+time in seconds since 1st January 1900.
```
Must be `1 January 1970`.

mschwan-phytec replied:
Sorry for the inconvenience, that was my fault.

mschwan-phytec commented on samples/net/sockets/sntp_client/README.rst:
```diff
@@ -0,0 +1,36 @@
+   :zephyr-app: samples/net/sockets/sntp_client
+   :board: qemu_x86
+   :goals: build
+   :compact: run
```
Remove the `run` in `:compact:`.

mschwan-phytec replied:
Again, my fault.

mschwan-phytec commented on samples/net/sockets/sntp_client/prj.conf:
```diff
@@ -26,3 +31,8 @@ CONFIG_LOG=y
 CONFIG_SNTP=y
+
+#CONFIG_NET_SOCKETS_LOG_LEVEL_DBG=y
+#CONFIG_SNTP_LOG_LEVEL_DBG=y
```
Either we enable the debug logging or leave it out completely.

pfalcon replied:
> Either we enable the debug logging or leave it out completely.

Why so strict?

mschwan-phytec replied:
I see no benefit in leaving the commented out configs in there. Shouldn't a sample be as minimal as possible?

pfalcon replied:
> Shouldn't a sample be as minimal as possible?

I don't think that "as minimal as possible" is a requirement for samples. They should be simple and easy to understand, yes. They also should be friendly to absolute newcomers (that's how people will get acquainted with Zephyr). The samples will misbehave, and users should have means to debug them at their fingertips (like, uncomment things, instead of reading 200+ pages manuals figuring out how to enable debug logging).

mschwan-phytec replied:
That is a valid point. But other samples do not have these config options commented out in them, ready to be activated for debugging. Why make *this* sample an exception now?

pfalcon replied:
> But other samples do not have

Some do.

> Why make this sample an exception now?

Because its author proposes this PR like that, and per above, we shouldn't discourage such practice.

nashif replied:
I do consider those as leftovers and they should not be in the config. What can  happen here?
- Most likely those were set to debug a certain issue, but they are not the right way or to debug and might be enabling debug just for some areas. 
- We might change the kconfig name and the sample would still have the old one which is not detected by CI because they are commented out
- It becomes a norm. New users would think this is required

How to debug the application should be rather explained in the documentation.

mschwan-phytec commented on samples/net/sockets/sntp_client/src/main.c:
```diff
@@ -0,0 +1,78 @@
+void resp_callback(struct sntp_ctx *ctx, int status,
+		   u64_t epoch_time)
+{
+	LOG_DBG("time: %lld", epoch_time);
```
AFAIK, this will not work. Only `printk()` can print 64 bit numbers. Using `LOG_DBG()` this could be done the following way, although it looks a bit hacky and prints the number as a hex value:
```c
LOG_DBG("time: %08x%08x", (u32_t) epoch_time >> 32, (u32_t) epoch_time & 0xffffffff);
```
Maybe there is some nicer way to print it as a decimal.

rveerama1 replied:
If logger doesn't support these kind of formats. May be open a issue to fix the problem in logger. For time being one can hack their samples. I will keep it as is.

mschwan-phytec replied:
This is a known limitation of the logging system and doubt it will be changed soon. See e.g. issue #12098. I may be wrong though. @nashif Any thoughts on this?

jukkar replied:
It might be a limitation atm, but to me this sounds like a serious bug if 64-bit values cannot be printed and should be fixed in the logger instead of adding some workarounds. Do you know if there is an issue created for this already?

jukkar replied:
Btw, the #12098 was for minimal libc, the issue is not related to logger.

mschwan-phytec commented on subsys/net/lib/sntp/sntp_pkt.h:
```diff
@@ -1,5 +1,6 @@
 /*
  * Copyright (c) 2017 Linaro Limited
+ * Copyright (c) 2019 Intel Corporation
```
I do not see any changes in this file, thus no copyright owner should be added IMHO.

mschwan-phytec commented on doc/zephyr.doxyfile.in:
```diff
@@ -810,6 +810,7 @@ EXCLUDE                = @ZEPHYR_BASE@/include/spi_legacy.h \
                          @ZEPHYR_BASE@/include/cmsis_rtos_v2/cmsis_os2.h \
                          @ZEPHYR_BASE@/include/net/coap.h \
                          @ZEPHYR_BASE@/include/net/coap_link_format.h \
+                         @ZEPHYR_BASE@/include/net/sntp.h \
```
Why is the SNTP documentation excluded here? (It is added to the list of Doxygen's `EXCLUDE`) We seem to be *replacing* and *not adding* a sockets based SNTP client API here.

mschwan-phytec reviewed (changes requested):
Some additional issues that I noticed.

rveerama1 commented on samples/net/sockets/sntp_client/prj.conf:
```diff
@@ -0,0 +1,36 @@
+# Network debug config
+CONFIG_NET_LOG=y
+CONFIG_LOG=y
+CONFIG_NET_SOCKETS_LOG_LEVEL_DBG=n
```
I have changed something like that today morning.

rveerama1 replied:
Any issues with that?

mschwan-phytec replied:
For me it implies that one must disable `NET_SOCKETS_LOG_LEVEL_DBG` for this sample to work, which seems wrong. I personally would still strip this line away but if the majority wants to leave something in then maybe use the commented out version, as used before.

rveerama1 replied:
> For me it implies that one must disable `NET_SOCKETS_LOG_LEVEL_DBG` for this sample to work,

really??

mschwan-phytec replied:
Sorry, if that sounded dumb. :confused: It was just my initial thought (especially from a new users point of view).

I think @nashif has a very good explanation, see his comment above.

dbkinder commented on samples/net/sockets/sntp_client/README.rst:
```diff
@@ -0,0 +1,29 @@
+Building and Running
+********************
+
+After the application was run it issues an SNTP request to the host and waits
```
Some grammar edits:
```
When the application runs, it issues an SNTP request to the host and waits
for a response. When the response is received, the current epoch time in
seconds, as well as the status code of the response (0 on success), is
printed.

rveerama1 replied:
@dbkinder thanks, addressed your comments.

mschwan-phytec commented on samples/net/sockets/sntp_client/CMakeLists.txt:
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.13.1)
+cmake_minimum_required(VERSION 3.8.2)
```
The minimum CMake version should be left at `3.13.1`.

mschwan-phytec commented on samples/net/sockets/sntp_client/src/main.c:
```diff
@@ -0,0 +1,78 @@
+
+#include <net/socket.h>
+#include <net/net_ip.h>
+#include <net/sntp.h>
```
Small improvement: Only `#include <net/sntp.h>` (and the logging system of course) is required for the sample to work.

mschwan-phytec reviewed (changes requested):
Two small things I noticed. Apart from that I'm fine with the changes now.

rveerama1 commented:
@mschwan-phytec Fixed.

mschwan-phytec reviewed (approved):
LGTM

pfalcon commented:
> So, given that no redesign phase was done for this PR, my preference would be to preserve the exact existing API. This includes reverting back to having void *user_data param for sntp_request(). 

@rveerama1, So, what's your stance on this? I see that the patchset has it removed like you did it originally on @tbursztyka's request and what I planned to revert.

pfalcon commented:
> If you really think it is mandatory in this PR, I will revert it.

I don't. It's your PR after all, people just asking why it's done like that and not done in another way. Given that we're already making a lot of "exceptions" with these hasty deprecations/removals, it's not that important indeed.

Removing my -1, but will need to look in more detail before +1'ing.

Thanks.

rveerama1 commented:
@dbkinder can you take a look again?

pfalcon commented:
So, reviewing the latest state, I can't refrain from a note that doing it like this is subideal. I mean, kill the original code, the in the next commit resurrect it under the authorship of different committer. If we decided to get rid of previous implementation (instead of cloning it), then the work should have been done by renaming and patching the original code.

Actually, let me request that first 2 commits ("Remove net_app based SNTP client library" and "Add socket based SNTP client library") were squashed together under the appropriate commit message, like "Rework SNTP client library to use sockets".

jukkar commented:
> Actually, let me request that first 2 commits ("Remove net_app based SNTP client library" and "Add socket based SNTP client library") were squashed together under the appropriate commit message, like "Rework SNTP client library to use sockets".

I do not see much point in that, it is quite convenient to review the stuff now.

pfalcon commented:
> I do not see much point in that, it is quite convenient to review the stuff now.

What's convenient in reviewing somebody else's code in a way as if it was written by @rveerama1 ?

pfalcon reviewed (changes requested):
Please squash first 2 commits together, updating the resulting commit message. Thanks.

jukkar commented:
> What's convenient in reviewing somebody else's code in a way as if it was written by @rveerama1 ?

Hmm, how information about who has written the code helps to review the code?

pfalcon commented:
> Hmm, how information about who has written the code helps to review the code?

But it was all explained above already - it helps if someone who changes just few lines of the code doesn't do that in a way which looks as if they wrote all the code from scratch.

See e.g. my above review suggestions - I'm asking @rveerama1 to change code comments which he didn't even write in the first place or changed in this PR. He just formatted his commits in such a way that they look as those comments coming from him.

nashif commented:
agree with @pfalcon here, lets squash the commits and move forward with this. We have those removal commits because we were trying to do things differently initially, now we are doing this all in one PR and replacing the old implementation, so it only make sense to have this a change commit rather than remove/add.

rveerama1 commented:
Squashed and addressed comments.

pfalcon reviewed (approved):
Thanks!

rveerama1 commented:
No changes, just rebased against master.

pfalcon commented:
@vanti: Please re-review (your -1 is blocking this PR).

vanti commented on samples/net/sockets/sntp_client/src/main.c:
```diff
@@ -0,0 +1,73 @@
+		       sizeof(struct sockaddr_in6));
+	if (rv < 0) {
+		LOG_ERR("Failed to initi sntp ctx: %d", rv);
+		goto end;
```
LGTM
